core/src/Dscp/Core/Aeson.hs

```diff
@@ -111,7 +111,7 @@ instance ToJSONKey Address
 deriving instance FromJSON GenAddressMap
 deriving instance ToJSON GenAddressMap
 instance FromJSON CommitteeSecret where
-    parseJSON = withText "CommitteeSecret" $ pure . CommitteeSecret . encodeUtf8
+    parseJSON = withText "CommitteeSecret" $ pure . mkCommitteeSecret . encodeUtf8
```

Comment:
This will lead not to graceful parser failure, but to a failure which would propagate all the way up to the IO monad. Same with other parsers. Make `mkCommitteeSecret` return `Either` instead and fail gracefully - this is a way to go, esp. given that the only places `CommitteeSecret` is constructed are some kinds of parsers.